### PR TITLE
Fixes rendering bug in Epoch.Time.Heatmap

### PR DESCRIPTION
**Pull request by rsandor**
At the top of the `Epoch.Time.Heatmap.redraw` there is a series of checks that ensures the chart has data before fully redrawing chart. It was erroneously inspecting the `@data` member when ensuring that the first layer had actual values to display.

### CODE REVIEW

**Comment by rsandor** [comment]
@jamesarosen - Can you take a look at this and make sure I am not mad? Also, if you have any suggestions for how to clean up that line to make it more readable, I am entirely comprised of ears.

**Comment by jamesarosen** [comment]
How about
```coffee
isNonEmptyArray = (ary) ->
  Epoch.isArray(ary) and ary.length > 0

redraw: ->
  return unless isNonEmptyArray(@data) and isNonEmptyArray(@data[0].values)
```

**Comment by rsandor** [comment]
Ooh, good call, could probably put that in Epoch.Util. Updates en route.

**Comment by rsandor** [comment]
@jamesarosen - That should do it, thoughts?

File: src/data.coffee [hunk 1]
```diff
@@ -69,7 +69,7 @@ Epoch.Data.Format.array = (->
     return result
 
   format = (data=[], options={}) ->
-    return [] unless Epoch.isArray(data) and data.length > 0
+    return [] unless Epoch.isNonEmptyArray(data)
```
Comment (jamesarosen) [inline src/data.coffee]: Seeing this double-negative tells me I suggested the wrong name. It's not the end of the world.

**Comment by jamesarosen** [comment]
+1

**Comment by rsandor** [comment]
Yeah, while non-empty is a double negative, I think it is used quite often to describe the test. An alternative would be to replace it with `not Epoch.isEmptyArray(ary)` but that just seems to make the test less clear. Should be good to go, from my perspective.

**Pull request merged**